build.sbt

```diff
@@ -100,6 +90,9 @@ lazy val scoverageSettings: Seq[Setting[_]] = Seq(
   parallelExecution in Test := false
 )
 
+def integrationComponentTestFilter(name: String): Boolean = (name startsWith "integration") || (name startsWith "component")
+def unitTestFilter(name: String): Boolean = name startsWith "unit"
```

Comment:
since you've added the unitTestFilter you may as well use it!
And then onPackage can be removed.